adapter/repository/in-memory-simple/src/test/java/org/mobilitydata/gtfsvalidator/db/InMemoryGtfsDataRepositoryTest.java

```diff
@@ -113,4 +114,53 @@ void getRouteByIdShouldReturnRelatedRoute() {
         assertEquals(mockRoute00, underTest.getRouteById("route id0"));
         assertEquals(mockRoute01, underTest.getRouteById("route id1"));
     }
+
+    @Test
+    void callToAddTransferShouldAddTransferToRepoAndReturnSameEntity() {
+        final Transfer mockTransfer = mock(Transfer.class);
+        final InMemoryGtfsDataRepository underTest = new InMemoryGtfsDataRepository();
+        when(mockTransfer.getFromStopId()).thenReturn("stop id 0");
+        when(mockTransfer.getToStopId()).thenReturn("stop id 1");
+
+        assertEquals(mockTransfer, underTest.addTransfer(mockTransfer));
+    }
+
+    @Test
+    void addSameTransferTwiceShouldReturnNull() {
+        final Transfer mockTransfer = mock(Transfer.class);
+        final InMemoryGtfsDataRepository underTest = new InMemoryGtfsDataRepository();
+        when(mockTransfer.getFromStopId()).thenReturn("stop id 0");
+        when(mockTransfer.getToStopId()).thenReturn("stop id 1");
+
+        underTest.addTransfer(mockTransfer);
+
+        assertNull(underTest.addTransfer(mockTransfer));
+    }
+
+    @Test
+    void addNullTransferShouldThrowIllegalArgumentException() {
+        final InMemoryGtfsDataRepository underTest = new InMemoryGtfsDataRepository();
+        final Exception exception = assertThrows(IllegalArgumentException.class,
+                () -> underTest.addTransfer(null));
+        assertEquals("Cannot add null transfer to data repository", exception.getMessage());
+    }
+
+    @Test
+    void getTransferByStopPairShouldReturnRelatedTransfer() {
+        final Transfer mockTransfer00 = mock(Transfer.class);
+        final Transfer mockTransfer01 = mock(Transfer.class);
+        final InMemoryGtfsDataRepository underTest = new InMemoryGtfsDataRepository();
+
+        when(mockTransfer00.getFromStopId()).thenReturn("stop id 0");
+        when(mockTransfer00.getToStopId()).thenReturn("stop id 1");
+
+        when(mockTransfer01.getFromStopId()).thenReturn("stop id 1");
+        when(mockTransfer01.getToStopId()).thenReturn("stop id 0");
+
+        underTest.addTransfer(mockTransfer00);
+        underTest.addTransfer(mockTransfer01);
+
+        assertEquals(mockTransfer00, underTest.getTransferByStopPair("stop id 0", "stop id 1"));
+        assertEquals(mockTransfer01, underTest.getTransferByStopPair("stop id 1", "stop id 0"));
```

Comment:
This test needs to be expanded to include multiple transfers per stop - it's missing a current bug when adding the record and more than one transfer for a stop exists.